src/main/scala/dxWDL/CompilerTopologicalSort.scala

```diff
@@ -0,0 +1,265 @@
+                 val newScatter = WdlRewrite.scatter(scatter,
+                    tsortASTnodes(scatter.children, cState, recursionDepth + 1))
+                 Utils.trace(cState.verbose, indent+"End recursion")
+                 newScatter.asInstanceOf[GraphNode]
+             }
+             case anyOtherNode => anyOtherNode.asInstanceOf[GraphNode]
+         }
+
+        // Create workflow graph sequence of edges for topological sorting
+        // Nodes in the graph (Scopes) can be a call, a scatter, or a declaration
+
+        // Create a mapping from a child of a scatter node to the root level scatter
+        // node it belongs to. This allows for a root level scatter node to be placed
+        // after its last dependency.
+
+        val scatterParent : Map[GraphNode, Scatter] = recursedNodes.map {
+            case scat: Scatter => scat.descendants.map { d => (d.asInstanceOf[GraphNode], scat) }
+            case _ => Set[(GraphNode, Scatter)]()
+        }.flatten.toMap
+
+        // Generate a list of edges corresponding to dependencies for this call
+        // e.g. suppose call x has dependencies a,b,c.  This procedure will generate:
+        // [(a,x), (b,x), (c,x)]
+        val edges : Set[(Scope, Scope)] = recursedNodes.map { gnode =>
+            val nodeDependencies : Vector[(Scope, Scope)] = (gnode.upstream).map { dependency =>
+               val dependencyActual : Scope =
+                   // If any node's parent is a descendant of a root
+                   // level scatter, use the scatter as the parent
+                   if (scatterParent.contains(dependency)) {
+                       scatterParent(dependency).asInstanceOf[Scope]
+                   }
+                   // Otherwise just use the actual parent
+                   else {
+                       dependency.asInstanceOf[Scope]
+                   }
+               (dependencyActual, gnode.asInstanceOf[Scope])
+            }.toVector
+
+            val descendantParents : Vector[(Scope, Scope)] = gnode match {
+                case scatter: Scatter => {
+                    // If a scatter's descendants have parents outside the scatter context,
+                    // map the scatter itself to them.
+                    val nodeDescendants = scatter.descendants.map { d => d.asInstanceOf[GraphNode] }
+                    val upstreamParents = nodeDescendants.map { d => d.upstream }.flatten.toSet
+                    ((upstreamParents -- nodeDescendants) - gnode).map{ parent =>
+                        // If the parent outside the scatter is also a member of a scatter, use it instead.
+                        if ( scatterParent.contains(parent) && gnode != scatterParent(parent)) {
+                            (scatterParent(parent), gnode)
+                        } else {
+                            (parent, gnode)
+                        }
+                    }.toVector
+                }
+                case anyOtherNode => Vector[(Scope, Scope)]()
```

Comment:
Elsewhere, I use the syntax:
case _ => Vector[(Scope, Scope)]()

I am fine with this style, though.